sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java

```diff
@@ -0,0 +1,162 @@
+import com.azure.cosmos.benchmark.linkedin.impl.models.GetRequestOptions;
+import com.azure.cosmos.benchmark.linkedin.impl.models.Result;
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The test implementation for GETs/pointed lookups against CosmosDB. It takes in
+ * the testData and submits upto N get operations at a time, leveraging the ExecutorService
+ * for the parallelism control.
+ *
+ * The Latency tracking and reposting is handled within the Accessor, and not in this method.
+ *
+ * TODO: Change how testData is stored. For the 10M document scenario, we should not hold it all in memory
+ */
+public class GetTestRunner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetTestRunner.class);
+    private static final boolean VALIDATE_RESULTS = false;
+
+    private final Configuration _configuration;
+    private final Accessor<Key, JsonNode> _accessor;
+    private final ExecutorService _executorService;
+
+    GetTestRunner(final Configuration configuration,
+        final CosmosAsyncClient client,
+        final MetricRegistry metricsRegistry) {
+        Preconditions.checkNotNull(configuration,
+            "The Workload configuration defining the parameters can not be null");
+        Preconditions.checkNotNull(client,
+            "Need a non-null client for setting up the Database and containers for the test");
+        Preconditions.checkNotNull(metricsRegistry,
+            "The MetricsRegistry can not be null");
+
+        _configuration = configuration;
+        _accessor = createAccessor(configuration, client, metricsRegistry);
+        _executorService = Executors.newFixedThreadPool(_configuration.getConcurrency());
+    }
+
+    public void run(Map<Key, ObjectNode> testData) {
+        final ArrayList<Key> keys = new ArrayList<>(testData.keySet());
+        Collections.shuffle(keys);
+        final long runStartTime = System.currentTimeMillis();
+        final AtomicLong successCount = new AtomicLong(0);
+        final AtomicLong errorCount = new AtomicLong(0);
+        long i = 0;
+        for (; BenchmarkHelper.shouldContinue(runStartTime, i, _configuration); i++) {
+            int index = (int) ((i % keys.size()) % Integer.MAX_VALUE);
+            final Key key = keys.get(index);
+            _executorService.submit(() -> runOperation(key, testData, successCount, errorCount));
```

Comment:
Here throughput RPS/QPS will not be equals to concurrency. We are using threads equals to concurrency , so lets say if we choose 50 threads, our throughput  will be much higher . No issue here but just mentioning this as per our offline conversation, so we are aware of our load.

Reply:
That's right. The actual QPS to the collection would be determined by the operation latency